lib/shoryuken/manager.rb

```diff
@@ -19,8 +19,11 @@ def initialize(fetcher, polling_strategy)
       @polling_strategy = polling_strategy
 
       @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+                                             execution_interval: HEARTBEAT_INTERVAL) { @pool.post { dispatch } if @dispatching.false? }
```

Comment:
@waynerobinson I know it isn't the most beautiful solution in the world. But that fix the thread leaking, I still need to play more with that. But I think the "final" fix will be something like that.